state-chain/runtime/src/lib.rs

```diff
@@ -2147,7 +2147,25 @@ impl_runtime_apis! {
 		}
 
 		fn cf_suspended_validators() -> Vec<(Offence, u32)> {
-			pallet_cf_reputation::Suspensions::<Runtime>::iter().map(|(key, elem)| (key, elem.len() as u32)).collect()
+			let current_block = <frame_system::Pallet<Runtime>>::block_number();
+			let suspended_for_keygen = match pallet_cf_validator::Pallet::<Runtime>::current_rotation_phase() {
+				pallet_cf_validator::RotationPhase::KeygensInProgress(rotation_state) |
+				pallet_cf_validator::RotationPhase::KeyHandoversInProgress(rotation_state) |
+				pallet_cf_validator::RotationPhase::ActivatingKeys(rotation_state) |
+				pallet_cf_validator::RotationPhase::NewKeysActivated(rotation_state) => { rotation_state.banned.len() as u32 },
+				_ => {0u32}
+			};
+			pallet_cf_reputation::Suspensions::<Runtime>::iter().map(|(key, elem)| {
+				if key == pallet_cf_threshold_signature::PalletOffence::FailedKeygen.into() {
+					return (key, suspended_for_keygen);
+				}
+				(key, elem.iter().filter_map(|(suspended_until, validator)| {
+					if *suspended_until > current_block {
+						return Some(validator);
+					}
+					None
+				}).collect::<BTreeSet<_>>().len() as u32)
+			}).collect()
```

Comment:
FYI there is a pallet method `Reputation::<T>::validators_suspended_for` that might make this a bit more readable.

(`pub fn validators_suspended_for(offences: &[T::Offence]) -> BTreeSet<T::ValidatorId>`)